### PR TITLE
Upgrade to Pytorch Lightning 1.5.5

### PR DESCRIPTION
**Pull request by ant0nsc**
Most changes were triggered by changes in the metrics API

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [ ] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [ ] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

**Comment by ant0nsc** [comment]
Putting that on hold. Upgrade leads to a 3% Dice drop on the Prostate model.

**Comment by ant0nsc** [comment]
Comparison with present code: 

|Structure	|NewMedianDice	|NewMeanDice	|OldMedianDice	|OldMeanDice|
|----|:---:|:---:|:---:|:---:|
|external	|0.996 |0.994734 |0.996 |0.99478|
|femur_l	|0.984	|0.983982	|0.984	|0.983826|
|femur_r	|0.984	|0.983734	|0.984	|0.983633|
|bladder	|0.972	|0.966211	|0.972	|0.966982|
|rectum	|0.904	|0.889266	|0.904	|0.889633|
|prostate	|0.866	|0.853092	|0.867	|0.853486|
|seminalvesicles	|0.797	|0.790826	|0.8	|0.792367|

Run on main: Prostate 5fold HD_ce452db4-5049-4393-a1ee-55d234cf2651
Run with new code: Prostate 5fold HD_1c475b3e-6923-465a-a5a2-38aacb946b87

**Review by mebristo — DISMISSED** [review]
Looks good to me!

File: InnerEye/ML/configs/classification/GlaucomaPublic.py [hunk 1]
```diff
@@ -37,6 +37,9 @@ def __init__(self, **kwargs: Any) -> None:
             train_batch_size=64,  # Batch size of 64 uses about 7GB of GPU memory
         )
         self.add_and_validate(kwargs)
+        # After Lightning 1.5.0 upgrade, torch complains that
+        # "avg_pool3d_backward_cuda does not have a deterministic implementation"
+        self.pl_deterministic = False
```
Comment (Shruthi42) [inline InnerEye/ML/configs/classification/GlaucomaPublic.py:42]: Could this go into the call to __init__ above?

**Review by Shruthi42 — APPROVED** [review]

**Review by mebristo — APPROVED** [review]

**Review by Shruthi42 — APPROVED** [review]

**Pull request merged**